compiler/testData/codegen/bytecodeText/forLoop/forInObjectArray.kt

```diff
@@ -0,0 +1,11 @@
+fun array() = arrayOfNulls<Any>(4)
+
+fun f() {
+    for (i in array()) {
+    }
```

Comment:
Please add some foo() call inside to avoid clashing with another optimizations in future:
```
 for (i in array()) {
    foo()
    }
```